### DISCUSSION TITLE
Field with @DateCreated annotation is deleted when I do an update

### DISCUSSION THREAD

**Discussion by Rutenioh** [micronaut-projects/micronaut-data#1485] [Q&A]
Hello, 
I am creating an API it saves an entity in a Mongo Db. This entity has a field with @DateCreated. In the post operation I call to the MongoRepository and the field is created.
But in the PUT operation, if I don't put the created field in the body, when I call to the update operation in MongoRepository, it deletes the field. 
I don't know if this is the proper behaviour with annotation or it is a bug.
Thank you in advance

**Comment by sdelamo** [comment]
do you have a project with a reproducible example?

**Reply by Rutenioh** [reply]
Yes, I attach to this comment
[manager-api.zip](https://github.com/micronaut-projects/micronaut-data/files/8638811/manager-api.zip)

You can check in the tests when I try to compare the created date

**Comment by dstepanov** [comment]
The update at the moment is implemented as `replaceOne` operation which is replacing the whole object.
